src/Symfony/Component/Mailer/Bridge/MailerSend/Transport/MailerSendApiTransport.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\MailerSend\Transport;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Mailer\Envelope;
+use Symfony\Component\Mailer\Exception\HttpTransportException;
+use Symfony\Component\Mailer\SentMessage;
+use Symfony\Component\Mailer\Transport\AbstractApiTransport;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mime\Email;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @see https://developers.mailersend.com/api/v1/email.html
+ */
+final class MailerSendApiTransport extends AbstractApiTransport
+{
+    private string $key;
+
+    public function __construct(string $key, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null, LoggerInterface $logger = null)
+    {
+        $this->key = $key;
+
+        parent::__construct($client, $dispatcher, $logger);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('mailersend+api://%s', $this->getEndpoint());
+    }
+
+    protected function doSendApi(SentMessage $sentMessage, Email $email, Envelope $envelope): ResponseInterface
+    {
+        $response = $this->client->request('POST', 'https://'.$this->getEndpoint().'/v1/email', [
+            'json' => $this->getPayload($email, $envelope),
+            'headers' => [
+                'Authorization' => 'Bearer '.$this->key,
+            ],
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+            $result = $response->getContent(false);
+            $headers = $response->getHeaders(false);
+        } catch (TransportExceptionInterface $e) {
+            throw new HttpTransportException('Could not reach the remote MailerSend server.', $response, 0, $e);
+        }
+
+        if (!empty($result)) {
+            try {
+                $result = json_decode($result, true, 512, \JSON_THROW_ON_ERROR);
```

Comment:
should use $response->toArray() instead

Reply:
[MailerSend API](https://developers.mailersend.com/api/v1/email.html#sending-queued) returns empty content if email was sent successfully and without any error. If sending email has some errors, MailerSend API returns [JSON content](https://developers.mailersend.com/api/v1/email.html#send-an-email).  For this reason, I cannot use `$response->toArray()` because it throws JsonException. The same exception is thrown for bad JSON format or if decoded JSON content is not an array.